spec/java/util/StringBuilder.main.lsl

```diff
@@ -0,0 +1,1022 @@
+        result = plane < ((MAX_CODE_POINT + 1) >>> 16);
+    }
+
+
+    proc _lowSurrogate (codePoint: int): char
+    {
+        result = ((codePoint & 1023) + MIN_LOW_SURROGATE) as char;
+    }
+
+
+    proc _highSurrogate (codePoint: int): char
+    {
+        result = ((codePoint >>> 10) + (MIN_HIGH_SURROGATE - (MIN_SUPPLEMENTARY_CODE_POINT >>> 10))) as char;
+    }
+
+    proc _appendCharSequence (seq: CharSequence): void
+    {
+        if (seq == null)
+            seq = "null";
+
+        val seqLength: int = action CALL_METHOD(seq, "length", []);
+        this.length += seqLength;
+
+        var i: int = 0;
+        action LOOP_FOR(i, 0, seqLength, +1, _appendCharSequence_loop(i, seq));
+    }
+
+
+    @Phantom proc _appendCharSequence_loop(i: int, seq: CharSequence): void
+    {
+        var currentChar: char = action CALL_METHOD(seq, "charAt", [i]);
+        this.storage += action OBJECT_TO_STRING(currentChar);
+    }
+
+
+    proc _appendString (str: String): void
+    {
+        if (str == null)
+            str = "null";
+        this.storage += str;
+        this.length = action CALL_METHOD(this.storage, "length", []);
+    }
+
+
+    proc _delete (start: int, end: int): void
+    {
+        val len: int = this.length - end + start;
+        var newString: array<char> = action ARRAY_NEW("char", len);
+
+        var i: int = 0;
+        var currentIndex: int = 0;
+        action LOOP_FOR(i, 0, this.length, +1, _deleteCharAt_loop(i, start, end, currentIndex, newString));
+
+        this.storage = action OBJECT_TO_STRING(newString);
+        this.length = len;
+    }
+
+
+    @Phantom proc _deleteCharAt_loop (i: int, start: int, end:int, currentIndex:int, newString: array<char>): void
+    {
+        if (i < start || i >= end)
+        {
+            val currentChar: char = action CALL_METHOD(this.storage, "charAt", [i]);
+            newString[currentIndex] = currentChar;
+            currentIndex += 1;
+        }
+    }
+
+
+    @AutoInline @Phantom proc _throwNPE (): void
+    {
+        action THROW_NEW("java.lang.NullPointerException", []);
+    }
+
+
+    proc _insertCharSequence (dstOffset: int, s: CharSequence, len: int, start: int, end: int): void
+    {
+        _checkRange(start, end, len);
+
+        val countInsertedElements: int = end - start;
+        val newStr: array<char> = action ARRAY_NEW("char", this.length + countInsertedElements);
+
+        var i: int = 0;
+        var currentIndex: int = start;
+        val endIndex: int = dstOffset + countInsertedElements;
+
+        action LOOP_FOR(i, 0, this.length, +1, _insertCharSequence_loop(i, dstOffset, endIndex, currentIndex, newStr, s));
+
+        this.storage = action OBJECT_TO_STRING(newStr);
+        this.length += countInsertedElements;
+    }
+
+
+    @Phantom proc _insertCharSequence_loop(i: int, dstOffset: int, endIndex: int, currentIndex:int, newStr: array<char>, s: CharSequence): void
+    {
+        if (i < dstOffset)
```

Comment:
@dog-m I really don't know hiw it will be more simple for USVM (I don't know any details about it internal realization);

But I have understood your hint;

Reply:
Fixed;